api/envoy/config/core/v3/base.proto

```diff
@@ -269,6 +269,9 @@ message HeaderValueOption {
   // Should the value be appended? If true (default), the value is appended to
   // existing values.
   google.protobuf.BoolValue append = 2;
+
+  // Whether the header should be modified when it is already present in the request or response.
+  bool skip_if_present = 3;
```

Comment:
I think this is fine, but I'm a bit concerned that we are placing `RouteConfiguraiton` specifics in the generic base type here. This is also used in `ext_authz`, where the field becomes irrelevant, and in the future it might also be used in other places. @envoyproxy/api-shepherds is this a reasonable concern or do we feel there is not likely to be further scope creep with `HeaderValueOption`?

Reply:
I feel like if we already have `append` here, `skip_if_present` is not really any different? So it's probably OK?